lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -1468,6 +1479,15 @@ private function validateAndCompleteTypedFieldMapping(array $mapping): array
                 ! isset($mapping['type'])
                 && ($type instanceof ReflectionNamedType)
             ) {
+                if (PHP_VERSION_ID >= 80100 && ! $type->isBuiltin() && enum_exists($type->getName(), false)) {
```

Comment:
@beberlei  I pushed a commit that added the `$type->isBuiltin()` part, hopefully that will make Psalm green :slightly_smiling_face: